src/controllers/__tests__/sessionLockController.test.ts

```diff
@@ -546,6 +548,138 @@ describe("SessionLockController#lockUserAuthentication", () => {
+  });
+});
+
+describe("SessionLockController#unlockUserAuthentication", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const aValidRequest = {
+    params: { fiscal_code: aFiscalCode },
+    body: { unlock_code: anUnlockCode },
+  };
+  const aValidRequestWithoutUnlockCode = {
+    params: { fiscal_code: aFiscalCode },
+  };
+
+  it.each`
+    title                                      | request
+    ${"request contains unlock code"}          | ${aValidRequest}
+    ${"request does NOT contains unlock code"} | ${aValidRequestWithoutUnlockCode}
+  `(
+    "should succeed releasing CF-unlockcode when user authentication is locked and $title",
+    async ({ request }) => {
+      const req = mockReq(request);
+      const res = mockRes();
+
+      getUserAuthenticationLockDataMock.mockReturnValueOnce(
+        TE.of(O.some(aNotReleasedData))
+      );
+
+      const response = await controller.unlockUserAuthentication(req);
+      response.apply(res);
+
+      expect(res.status).toHaveBeenCalledWith(204);
+
+      expect(getUserAuthenticationLockDataMock).toHaveBeenCalledWith(
+        aFiscalCode
+      );
+      expect(unlockUserAuthenticationMock).toHaveBeenCalledWith(
+        aFiscalCode,
+        anUnlockCode
+      );
+    }
+  );
+
+  it.each`
+    title                                      | request
+    ${"request contains unlock code"}          | ${aValidRequest}
+    ${"request does NOT contains unlock code"} | ${aValidRequestWithoutUnlockCode}
+  `(
+    "should fail releasing CF-unlockcode when the user authentication is NOT locked and title",
+    async ({ request }) => {
+      const req = mockReq(request);
+      const res = mockRes();
+
+      const response = await controller.unlockUserAuthentication(req);
+      response.apply(res);
+
+      expect(res.status).toHaveBeenCalledWith(500);
+
+      expect(getUserAuthenticationLockDataMock).toHaveBeenCalledWith(
+        aFiscalCode
+      );
+      expect(unlockUserAuthenticationMock).not.toHaveBeenCalled();
+    }
+  );
+
+  it("should fail releasing CF-unlockcode when the user authentication is locked with another unlock code", async () => {
+    const req = mockReq({
+      ...aValidRequestWithoutUnlockCode,
+      body: { unlock_code: "987654321" },
+    });
+    const res = mockRes();
+
+    getUserAuthenticationLockDataMock.mockReturnValueOnce(
+      TE.of(O.some(aNotReleasedData))
+    );
+
+    const response = await controller.unlockUserAuthentication(req);
+    response.apply(res);
+
+    expect(response).toEqual(
+      expect.objectContaining({
+        detail: "Internal server error: Invalid unlock code",
+      })
+    );
+
+    expect(res.status).toHaveBeenCalledWith(500);
+
+    expect(getUserAuthenticationLockDataMock).toHaveBeenCalledWith(aFiscalCode);
+    expect(unlockUserAuthenticationMock).not.toHaveBeenCalled();
+  });
+
+  it("should return InternalServerError when an error occurred releasing authentication lock", async () => {
```

Comment:
I think we are missing a test almost identical to this, where `getUserAuthenticationLockDataMock` returns an `Error`